only show build settings for build commands

dhellmann opened the pull request:
Fixes #440

rd4398 reviewed (commented):
The changes look good! Just a couple of things:

- Looks like the `mypy` checks are failing
- The linked issue mentions similar behavior for the `graph` based commands as well. However, I did not see any changes related to this. Can the similar changes be applied there? or did I miss anything?

dhellmann commented:
> The changes look good! Just a couple of things:
> 
> * Looks like the `mypy` checks are failing

I fixed that in a separate PR, so CI should be green now.

> * The linked issue mentions similar behavior for the `graph` based commands as well. However, I did not see any changes related to this. Can the similar changes be applied there? or did I miss anything?

I implemented the check by looking for a flag to be set, and defaulting to False if the flag is not found. That way I only had to enable the flag on a few commands where we do want the output. I'll add a comment to that effect where the flag is being checked.
